filebeat/prospector/log/stdin.go

```diff
@@ -2,17 +2,28 @@ package log
 
 import (
 	"os"
-
-	"github.com/elastic/beats/filebeat/harvester/source"
 )
 
 // Stdin reads all incoming traffic from stdin and sends it directly to the output
 
 func (h *Harvester) openStdin() error {
-	h.file = source.Pipe{File: os.Stdin}
+	h.source = Pipe{File: os.Stdin}
 
 	var err error
-	h.encoding, err = h.encodingFactory(h.file)
+	h.encoding, err = h.encodingFactory(h.source)
 
 	return err
 }
+
+// restrict file to minimal interface of FileSource to prevent possible casts
+// to additional interfaces supported by underlying file
+type Pipe struct {
+	File *os.File
+}
+
+func (p Pipe) Read(b []byte) (int, error) { return p.File.Read(b) }
+func (p Pipe) Close() error               { return p.File.Close() }
+func (p Pipe) Name() string               { return p.File.Name() }
+func (p Pipe) Stat() (os.FileInfo, error) { return p.File.Stat() }
+func (p Pipe) Continuable() bool          { return false }
+func (p Pipe) HasState() bool             { return false }
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Pipe.HasState should have comment or be unexported